lib/Remote/RemoteWebElement.php

```diff
@@ -498,26 +498,38 @@ protected function upload($local_file)
             throw new WebDriverException('You may only upload files: ' . $local_file);
         }
 
+        $temp_zip_path = $this->createTemporaryZipArchive($local_file);
+
+        $remote_path = $this->executor->execute(
+            DriverCommand::UPLOAD_FILE,
+            ['file' => base64_encode(file_get_contents($temp_zip_path))]
+        );
+
+        unlink($temp_zip_path);
+
+        return $remote_path;
+    }
+
+    /**
+     * @param string $fileToZip
+     * @return string
+     */
+    protected function createTemporaryZipArchive($fileToZip)
+    {
         // Create a temporary file in the system temp directory.
-        $temp_zip = tempnam(sys_get_temp_dir(), 'WebDriverZip');
+        // Intentionally do not use `tempnam()`, as it creates empty file which zip extension may not handle.
+        $tempZipPath = sys_get_temp_dir() . '/' . uniqid('WebDriverZip', false);
```

Comment:
why not using `DIRECTORY_SEPARATOR` ?

Reply:
Looks like this was changed by @OndraM, it should be the DIRECTORY_SEPARATOR constant used here because it will break on systems running windows.

Reply:
Yes, I changed it intentionally, as  DIRECTORY_SEPARATOR is not needed, as "/" works on both unix and windows systems.

https://dev.to/c33s/always-use--as-directory-seperator-in-php-43l7